src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -33,22 +48,31 @@ const HeatmapPlot = (props) => {
   const loadingGenes = useSelector((state) => state.genes.expression.loading);
   const selectedGenes = useSelector((state) => state.genes.expression.views[COMPONENT_TYPE]?.data);
 
-  const [vegaData, setVegaData] = useState(null);
-  const [vegaSpec, setVegaSpec] = useState(spec);
+  const [viewState, setViewState] = useState({ zoom: 0, target: [0, 0] });
+  const [heatmapData, setHeatmapData] = useState(null);
+  const [highlightedTrackData, setHighlightedTrackData] = useState(null);
+
   const [isHeatmapGenesLoading, setIsHeatmapGenesLoading] = useState(false);
-  const currentHeatmapSettings = useRef();
 
+  const [geneHighlight, setGeneHighlight] = useState(null);
+  const [cellHighlight, setCellHighlight] = useState(null);
+
+  const [maxCells, setMaxCells] = useState(1000);
+  const [vitessceData, setVitessceData] = useState(null);
+
+  const cellCoordinatesRef = useRef({ x: 200, y: 300 });
+  const currentHeatmapSettingsRef = useRef(null);
```

Comment:
shouldn't the default be equal to some default settings, not `null`? Do these settings refer to the settings that the user selects on the top right corner of the marker heatmap window?

Reply:
Yes, they are those settings.
Usually we store the default settings in redux, so from the first `useSelector` this is filled in with the actual default settings

Reply:
from what I can see, this `useRef` is not used to have those settings accessed and used, it is just to compare with previous settings and know whether to repopulate the heatmap data or not

Reply:
I think this useRef could be removed now that I look at it, it is used to see if this changed on a `useSelector` trigger:
`const heatmapSettings = useSelector((state) => state.componentConfig[COMPONENT_TYPE]?.config) || {};`

but actually this should be enough to replace the `useRef` functionality:
`const heatmapSettings = useSelector((state) => state.componentConfig[COMPONENT_TYPE]?.config, _.isEqual) || {};`

Reply:
sounds good, let's remove it if we don't need it then!